tapgarden/custodian.go

```diff
@@ -865,6 +870,9 @@ func (c *Custodian) setReceiveCompleted(event *address.Event,
 		Index: lastProof.InclusionProof.OutputIndex,
 	}
 
+	// TODO(ffranr): Remove event from local cache after
+	//  AddrBook.CompleteEvent call.
+
```

Comment:
We should only publish the "receive complete" status notification to subscribers after the call to `c.cfg.AddrBook.CompleteEvent`. Encase anything fails beforehand.